### PR TITLE
repos: do not encode token to an empty `repo.URL` for GitHub

### PR DESCRIPTION
**Pull request by unknwon**
This PR is related to COREAPP-73 but not necessarily fixes the root cause of it (i.e. some GitHub repos are having empty `repo.metadata` column on dotcom).

This PR fixes the root cause of COREAPP-74 in fact, which we're trying to encode tokens into an empty `repo.URL` and resulted as invalid clone URL.

### CODE REVIEW

File: internal/repos/clone_url.go [hunk 1]
```diff
@@ -66,10 +66,8 @@ func CloneURL(kind, config string, repo *types.Repo) (string, error) {
 		if r, ok := repo.Metadata.(*extsvc.OtherRepoMetadata); ok {
 			return otherCloneURL(repo, r), nil
 		}
-	default:
-		return "", errors.Errorf("unknown external service kind %q for repo %d", kind, repo.ID)
 	}
-	return "", errors.Errorf("unknown repo.Metadata type %T for repo %d", repo.Metadata, repo.ID)
+	return "", errors.Errorf("unknown external service kind %q for repo %d", kind, repo.ID)
```
Comment (unknwon) [inline internal/repos/clone_url.go:70]: The old return will never be reached because `default` catches all unmatches.

Comment (ryanslade) [reply internal/repos/clone_url.go:70]: There's a chance that we match on `parsed.(type)` but then fail when we try to convert `repo.MetaData`

For example, if one of the `ok` bools are false.

File: internal/repos/clone_url_test.go [hunk 2]
```diff
@@ -184,7 +195,7 @@ func TestGithubCloneURLs(t *testing.T) {
 	}
 }
 
-func TestGitlabCloneURLs(t *testing.T) {
+func TestGitLabCloneURLs(t *testing.T) {
```
Comment (ryanslade) [inline internal/repos/clone_url_test.go:198]: 😄

**Review by ryanslade — APPROVED** [review]
Approve to unblock but I think we need to change the switch back.

File: internal/repos/clone_url.go (hunk 1 above)
Comment (unknwon) [reply internal/repos/clone_url.go:70]: You're right, I'll revert this part.

**Pull request merged**